gcloud/monitoring/group.py

```diff
@@ -0,0 +1,507 @@
+            self.client.connection.api_request(method='GET', path=self.path)
+        except NotFound:
+            return False
+        else:
+            return True
+
+    def reload(self):
+        """Sync local group information via a ``GET`` request.
+
+        .. warning::
+
+            This will overwrite any local changes you've made and not saved
+            via :meth:`update`.
+        """
+        info = self.client.connection.api_request(method='GET', path=self.path)
+        self._init_from_dict(info)
+
+    def update(self):
+        """Update the group via a ``PUT`` request."""
+        info = self.client.connection.api_request(
+            method='PUT', path=self.path, data=self._to_dict())
+        self._init_from_dict(info)
+
+    def delete(self):
+        """Delete the group via a ``DELETE`` request.
+
+        Example::
+
+            >>> group = client.group('1234')
+            >>> group.delete()
+
+        Only the ``client`` and ``name`` attributes are used.
+
+        .. warning::
+
+            This method will fail for groups that have one or more children
+            groups.
+        """
+        self.client.connection.api_request(method='DELETE', path=self.path)
+
+    def parent(self):
+        """Returns the parent group of this group via a ``GET`` request.
+
+        :rtype: :class:`Group` or None
+        :returns: The parent of the group.
+        """
+        if not self.parent_name:
+            return None
+        return self._fetch(self.client, self.parent_id)
+
+    def children(self):
+        """Lists all children of this group via a ``GET`` request.
+
+        Returns groups whose parent_name field contains the group name. If no
+        groups have this parent, the results are empty.
+
+        :rtype: list of :class:`~gcloud.monitoring.group.Group`
+        :returns: A list of group instances.
+        """
+        return self._list(self.client, children_of_group=self.name)
+
+    def ancestors(self):
+        """Lists all ancestors of this group via a ``GET`` request.
+
+        The groups are returned in order, starting with the immediate parent
+        and ending with the most distant ancestor. If the specified group has
+        no immediate parent, the results are empty.
+
+        :rtype: list of :class:`~gcloud.monitoring.group.Group`
+        :returns: A list of group instances.
+        """
+        return self._list(self.client, ancestors_of_group=self.name)
+
+    def descendants(self):
+        """Lists all descendants of this group via a ``GET`` request.
+
+        This returns a superset of the results returned by the :meth:`children`
+        method, and includes children-of-children, and so forth.
+
+        :rtype: list of :class:`~gcloud.monitoring.group.Group`
+        :returns: A list of group instances.
+        """
+        return self._list(self.client, descendants_of_group=self.name)
+
+    def members(self, filter_string=None, end_time=None, start_time=None):
```

Comment:
Along with this change, it would also make sense to rename the 4 methods before it:

parent -> fetch_parent
children -> list_children
ancestors -> list_ancestors
descendants -> list_descendants

Does that sound good?